File: lib/src/features/auth/data/dto/user_output_dto.dart
class UserOutputDto {
  const UserOutputDto(this.id, this.fullname, this.birthdate, this.address,
      this.email, this.photo, this.token);

  final int id;
  final String fullname;
  final String birthdate;
  final String address;
  final String email;
  final String photo;
  final String token;

  factory UserOutputDto.fromJsonToken(Map<String, dynamic> json, String token) => UserOutputDto(
      json['id'],
      json['fullname'],
      json['birthdate'],
      json['address'],
      json['email'],
      json['photo'],
      token);

  factory UserOutputDto.fromJson(Map<String, dynamic> json) => UserOutputDto(
      json['id'],
      json['fullname'],
      json['birthdate'],
      json['address'],
      json['email'],
      json['photo'],
      json['token']);

  Map<String, dynamic> toJson() => {
        'id': id,
        'fullname': fullname,
        'birthdate': birthdate.replaceAll('/', '-'),
        'address': address,
        'email': email,
        'photo': photo,
        'token': token
      };
}

File: api/src/main/java/br/com/fakenews/dto/UserOutputDto.java
package br.com.fakenews.dto;

import java.util.Date;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserOutputDto {
	
	@NotNull
	private Long id;

	@NotBlank
	private String email;
	
	@NotBlank
	private String fullname;
	
	@NotNull
	private Date birthdate;
	
	private String photo;
	
	private String address;
}
